packages/api/app/Models/User.js

```diff
@@ -132,6 +132,32 @@ class User extends Model {
 	isVerified() {
 		return this.status === 'verified';
 	}
+
+	/**
+	 * Runs the user query with the provided filters.
+	 *
+	 * @param {object} query The query object.
+	 * @param {object} filters The query filters
+	 *
+	 * @returns {object}
+	 */
+	static scopeWithFilters(query, filters) {
+		const technologyId = Number(filters.technologyId);
+
+		if (technologyId) {
+			query
+				.whereHas('bookmarks', (builder) => {
+					builder.where({ technology_id: technologyId });
+				})
+				.with('bookmarks', (builder) => {
+					builder.where({ technology_id: technologyId });
+				});
+		} else {
+			query.with('bookmarks');
+		}
```

Comment:
```suggestion
      query.with('bookmarks');
		if (technologyId) {
			query
				.whereHas('bookmarks', (builder) => {
					builder.where({ technology_id: technologyId });
				});
		
```

Não testei mas tenho pra mim que a consulta acima funciona! A query vai adicionar todos os bookmarks mas o queryHas deve fazer a filtragem.